sklearn/preprocessing/_encoders.py

```diff
@@ -675,16 +675,10 @@ def _legacy_transform(self, X):
 
     def _transform_new(self, X):
         """New implementation assuming categorical input"""
-        X_temp = check_array(X, dtype=None)
-        if not hasattr(X, 'dtype') and np.issubdtype(X_temp.dtype, np.str_):
-            X = check_array(X, dtype=np.object)
-        else:
-            X = X_temp
-
-        n_samples, n_features = X.shape
-
```

Comment:
Maybe it would be useful to add a comment saying that input validation is done in self._transform